Build fails on ESP32 when enabling websocket client API

**Describe the bug**
When enabling `CONFIG_WEBSOCKET_CLIENT`, compilation fails on ESP32 presumably due to `CONFIG_BASE64` being enabled by `CONFIG_WEBSOCKET_CLIENT`, of which the functions are already defined in the Espressif HAL module.

Target platform: Xtensa esp32

**To Reproduce**
Enable `CONFIG_WEBSOCKET_CLIENT` in `prj.conf` of any project, and compile for `esp32` with `west build -b esp32 <project>`

**Expected behavior**
Build for esp32 to succeed with the websocket client API. 

**Impact**
Currently stalls development. Could write my own implementation of websocket client if necessary

**Logs and console output**
```
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr/zephyr_pre0.map 
: && ccache /nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/xtensa-espressif_esp32_zephyr-elf-g++  -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/xtensa/core/offsets/offsets.c.obj  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/xtensa/core/libarch__xtensa__core.a  zephyr/lib/libc/newlib/liblib__libc__newlib.a  zephyr/subsys/net/libsubsys__net.a  zephyr/subsys/net/l2/ethernet/libsubsys__net__l2__ethernet.a  zephyr/subsys/net/l2/wifi/libsubsys__net__l2__wifi.a  zephyr/subsys/net/ip/libsubsys__net__ip.a  zephyr/subsys/net/lib/dns/libsubsys__net__lib__dns.a  zephyr/subsys/net/lib/http/libsubsys__net__lib__http.a  zephyr/subsys/net/lib/websocket/libwebsocket_client.a  zephyr/subsys/random/libsubsys__random.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/wifi/libdrivers__wifi.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/entropy/libdrivers__entropy.a  zephyr/drivers/timer/libdrivers__timer.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  modules/mbedtls/libmodules__mbedtls.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/12.1.0"  -L/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -no-pie  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -lstdc++  -lm  -Wl,-lc  -L"/nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/xtensa-espressif_esp32_zephyr-elf"/lib/.  -Wl,-lgcc  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/src/linker/esp32.rom.alias.ld  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.ld  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.api.ld  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.libgcc.ld  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.redefined.ld  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../../components/soc/esp32/ld/esp32.peripherals.ld  -Wl,--wrap=k_calloc  -Wl,--wrap=k_malloc  -lc  -lnet80211  -lcore  -lpp  -lcoexist  -lphy  -lrtc  -lc  -lnet80211  -lcore  -lpp  -lcoexist  -lphy  -lrtc  -lgcc  -L/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../blobs/lib/esp32 && cd /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr && /nix/store/vh76h64xifdiylk1dfc9s6mgpzazngxs-cmake-3.24.3/bin/cmake -E echo
/nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/12.1.0/../../../../xtensa-espressif_esp32_zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(base64.c.obj): in function `base64_encode':
/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/wpa_supplicant/src/utils/base64.c:166: multiple definition of `base64_encode'; zephyr/libzephyr.a(base64.c.obj):/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/lib/os/base64.c:62: first defined here
/nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/12.1.0/../../../../xtensa-espressif_esp32_zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(base64.c.obj): in function `base64_decode':
/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/wpa_supplicant/src/utils/base64.c:188: multiple definition of `base64_decode'; zephyr/libzephyr.a(base64.c.obj):/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/lib/os/base64.c:126: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /nix/store/vh76h64xifdiylk1dfc9s6mgpzazngxs-cmake-3.24.3/bin/cmake --build /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build
```

**Environment (please complete the following information):**
 - OS: Linux (NixOS)
 - Toolchain: Zephyr SDK 0.15.2, xtensa-espressif_esp32_zephyr-elf
 - Commit SHA: `b8f8c3c10eda6d1529d0e616b24ba9719efab2c8`

Using `samples/net/wifi` as example; Add `CONFIG_WEBSOCKET_CLIENT=y` to `samples/net/wifi/prj.conf`, then build with `west build -b esp32 samples/net/wifi`

My full build output of it:
```
$ west build -p auto -b esp32 samples/net/wifi          
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/samples/net/wifi
-- Found Python3: /nix/store/s16n3im3ajnbzamp6yzzp15cwxqn8pbr-python3-3.10.8/bin/python3.10 (found suitable exact version "3.10.8") found components: Interpreter 
-- Cache files will be written to: /home/ricardo/.cache//zephyr
-- Zephyr version: 3.2.99 (/home/ricardo/Documents/Programming/zephyr-workspace/zephyr)
-- Found west (found suitable version "0.14.0", minimum required is "0.7.1")
-- Board: esp32
-- Found host-tools: zephyr 0.15.2 (/nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2)
-- Found toolchain: zephyr 0.15.2 (/nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2)
-- Found Dtc: /nix/store/pq5gybjkrnwpkhcc9zrvf1ib2fyk50xh-dtc-1.6.1/bin/dtc (found suitable version "1.6.1", minimum required is "1.4.6") 
-- Found BOARD.dts: /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/boards/xtensa/esp32/esp32.dts
-- Found devicetree overlay: /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/samples/net/wifi/boards/esp32.overlay
-- Generated zephyr.dts: /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr/dts.cmake
/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr/zephyr.dts:37.39-44.5: Warning (interrupt_provider): /soc/interrupt-controller@3ff00104: Missing #address-cells in interrupt provider
Parsing /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/Kconfig
Loaded configuration '/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/boards/xtensa/esp32/esp32_defconfig'
Merged configuration '/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/samples/net/wifi/prj.conf'
Merged configuration '/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/samples/net/wifi/boards/esp32.conf'
Configuration saved to '/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr/.config'
Kconfig header saved to '/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- The ASM compiler identification is GNU
-- Found assembler: /nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/xtensa-espressif_esp32_zephyr-elf-gcc
CMake Warning at /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/CMakeLists.txt:824 (message):
  No SOURCES given to Zephyr library: drivers__ethernet

  Excluding target from build.


-- Configuring done
-- Generating done
-- Build files have been written to: /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build
-- west build: building application
[1/407] Preparing syscall dependency handling

[14/407] Generating include/generated/version.h
-- Zephyr version: 3.2.99 (/home/ricardo/Documents/Programming/zephyr-workspace/zephyr), build: zephyr-v3.2.0-2647-gb8f8c3c10eda
[30/407] Performing configure step for 'EspIdfBootloader'
-- Found Git: /nix/store/1im1jlvkqfzwdwyps9nwychkl638ypz7-git-2.38.1/bin/git (found version "2.38.1") 
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- The ASM compiler identification is GNU
-- Found assembler: /nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/xtensa-espressif_esp32_zephyr-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/xtensa-espressif_esp32_zephyr-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/xtensa-espressif_esp32_zephyr-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32
-- Project sdkconfig file /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/esp-idf/build/sdkconfig
-- Adding linker script /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/soc/esp32/ld/esp32.peripherals.ld
-- Adding linker script /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/esp_rom/esp32/ld/esp32.rom.ld
-- Adding linker script /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/esp_rom/esp32/ld/esp32.rom.api.ld
-- Adding linker script /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/esp_rom/esp32/ld/esp32.rom.libgcc.ld
-- Adding linker script /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/esp_rom/esp32/ld/esp32.rom.newlib-funcs.ld
-- Adding linker script /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/bootloader/subproject/main/ld/esp32/bootloader.ld
-- Adding linker script /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/bootloader/subproject/main/ld/esp32/bootloader.rom.ld
-- Components: bootloader bootloader_support efuse esp32 esp_common esp_hw_support esp_rom esp_system esptool_py freertos hal log main micro-ecc newlib partition_table soc spi_flash xtensa
-- Component paths: /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/bootloader /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/bootloader_support /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/efuse /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/esp32 /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/esp_common /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/esp_hw_support /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/esp_rom /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/esp_system /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/esptool_py /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/freertos /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/hal /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/log /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/bootloader/subproject/main /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/bootloader/subproject/components/micro-ecc /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/newlib /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/partition_table /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/soc /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/spi_flash /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/xtensa
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/esp-idf/build/bootloader
[230/407] Performing build step for 'EspIdfBootloader'
[1/97] Generating project_elf_src_esp32.c
[2/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/soc_include_legacy_warn.c.obj
[3/97] Building C object CMakeFiles/bootloader.elf.dir/project_elf_src_esp32.c.obj
[4/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/adc_periph.c.obj
[5/97] Building C object esp-idf/hal/CMakeFiles/__idf_hal.dir/mpu_hal.c.obj
[6/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/dac_periph.c.obj
[7/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/lldesc.c.obj
[8/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/gpio_periph.c.obj
[9/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/i2c_periph.c.obj
[10/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/interrupts.c.obj
[11/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/i2s_periph.c.obj
[12/97] Building C object esp-idf/hal/CMakeFiles/__idf_hal.dir/cpu_hal.c.obj
[13/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/lcd_periph.c.obj
[14/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/pcnt_periph.c.obj
[15/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/ledc_periph.c.obj
[16/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/mcpwm_periph.c.obj
[17/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/rmt_periph.c.obj
[18/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/rtc_io_periph.c.obj
[19/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/sdmmc_periph.c.obj
[20/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/timer_periph.c.obj
[21/97] Building C object esp-idf/hal/CMakeFiles/__idf_hal.dir/wdt_hal_iram.c.obj
[22/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/sigmadelta_periph.c.obj
[23/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/spi_periph.c.obj
[24/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/uart_periph.c.obj
[25/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/sdio_slave_periph.c.obj
[26/97] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32/touch_sensor_periph.c.obj
[27/97] Building C object esp-idf/spi_flash/CMakeFiles/__idf_spi_flash.dir/esp32/spi_flash_rom_patch.c.obj
[28/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_clock_init.c.obj
[29/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_mem.c.obj
[30/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_flash.c.obj
[31/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_random.c.obj
[32/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_common_loader.c.obj
[33/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_common.c.obj
[34/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_random_esp32.c.obj
[35/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/secure_boot.c.obj
[36/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/flash_encrypt.c.obj
[37/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/flash_partitions.c.obj
[38/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_utility.c.obj
[39/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp_image_format.c.obj
[40/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/flash_qio_mode.c.obj
[41/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_clock_loader.c.obj
[42/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_console_loader.c.obj
[43/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_efuse_esp32.c.obj
[44/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_flash_config_esp32.c.obj
[45/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_init.c.obj
[46/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp32/bootloader_soc.c.obj
[47/97] Building C object esp-idf/micro-ecc/CMakeFiles/__idf_micro-ecc.dir/uECC_verify_antifault.c.obj
[48/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_panic.c.obj
[49/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_console.c.obj
[50/97] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/esp32/esp_efuse_table.c.obj
[51/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp32/bootloader_sha.c.obj
[52/97] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp_efuse_fields.c.obj
[53/97] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/esp32/esp_efuse_fields.c.obj
[54/97] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/esp32/esp_efuse_utility.c.obj
[55/97] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp_efuse_api.c.obj
[56/97] Building C object esp-idf/esp_system/CMakeFiles/__idf_esp_system.dir/esp_err.c.obj
[57/97] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp_efuse_utility.c.obj
[58/97] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp_efuse_api_key_esp32.c.obj
[59/97] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/compare_set.c.obj
[60/97] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/cpu_util.c.obj
[61/97] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_init.c.obj
[62/97] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_pm.c.obj
[63/97] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_clk.c.obj
[64/97] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp32/bootloader_esp32.c.obj
[65/97] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/chip_info.c.obj
[66/97] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_sleep.c.obj
[67/97] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_clk_init.c.obj
[68/97] Building C object esp-idf/xtensa/CMakeFiles/__idf_xtensa.dir/eri.c.obj
[69/97] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_crc.c.obj
[70/97] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_wdt.c.obj
[71/97] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32/rtc_time.c.obj
[72/97] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_sys.c.obj
[73/97] Building C object esp-idf/xtensa/CMakeFiles/__idf_xtensa.dir/xt_trax.c.obj
[74/97] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_tjpgd.c.obj
[75/97] Building ASM object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_longjmp.S.obj
[76/97] Building C object esp-idf/esp_common/CMakeFiles/__idf_esp_common.dir/src/esp_err_to_name.c.obj
[77/97] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_uart.c.obj
[78/97] Building C object esp-idf/log/CMakeFiles/__idf_log.dir/log_noos.c.obj
[79/97] Building C object esp-idf/main/CMakeFiles/__idf_main.dir/bootloader_start.c.obj
[80/97] Building C object esp-idf/log/CMakeFiles/__idf_log.dir/log_buffers.c.obj
[81/97] Building C object esp-idf/log/CMakeFiles/__idf_log.dir/log.c.obj
[82/97] Linking C static library esp-idf/log/liblog.a
[83/97] Linking C static library esp-idf/esp_rom/libesp_rom.a
[84/97] Linking C static library esp-idf/esp_common/libesp_common.a
[85/97] Linking C static library esp-idf/xtensa/libxtensa.a
[86/97] Linking C static library esp-idf/esp_hw_support/libesp_hw_support.a
[87/97] Linking C static library esp-idf/esp_system/libesp_system.a
[88/97] Linking C static library esp-idf/efuse/libefuse.a
[89/97] Linking C static library esp-idf/bootloader_support/libbootloader_support.a
[90/97] Linking C static library esp-idf/spi_flash/libspi_flash.a
[91/97] Linking C static library esp-idf/micro-ecc/libmicro-ecc.a
[92/97] Linking C static library esp-idf/soc/libsoc.a
[93/97] Linking C static library esp-idf/hal/libhal.a
[94/97] Linking C static library esp-idf/main/libmain.a
[95/97] Linking C executable bootloader.elf
[96/97] Generating binary image from built executable
esptool.py v3.3
Creating esp32 image...
Merged 1 ELF section
Successfully created esp32 image.
Generated /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/esp-idf/build/bootloader/bootloader.bin
[97/97] cd /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/esp-idf/build/bootloader/esp-idf/esptool_py && /nix/store/s16n3im3ajnbzamp6yzzp15cwxqn8pbr-python3-3.10.8/bin/python3.10 /home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/partition_table/check_sizes.py --offset 0x8000 bootloader 0x1000 /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/esp-idf/build/bootloader/bootloader.bin
Bootloader binary size 0x6520 bytes. 0xae0 bytes (10%) free.
[397/407] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr/zephyr_pre0.map 
: && ccache /nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/xtensa-espressif_esp32_zephyr-elf-gcc  -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/xtensa/core/offsets/offsets.c.obj  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/xtensa/core/libarch__xtensa__core.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/subsys/net/libsubsys__net.a  zephyr/subsys/net/l2/ethernet/libsubsys__net__l2__ethernet.a  zephyr/subsys/net/l2/wifi/libsubsys__net__l2__wifi.a  zephyr/subsys/net/ip/libsubsys__net__ip.a  zephyr/subsys/net/lib/http/libsubsys__net__lib__http.a  zephyr/subsys/net/lib/websocket/libwebsocket_client.a  zephyr/subsys/random/libsubsys__random.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/wifi/libdrivers__wifi.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/entropy/libdrivers__entropy.a  zephyr/drivers/timer/libdrivers__timer.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  modules/mbedtls/libmodules__mbedtls.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/12.1.0"  -L/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -no-pie  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/src/linker/esp32.rom.alias.ld  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.ld  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.api.ld  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.libgcc.ld  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../../components/esp_rom/esp32/ld/esp32.rom.redefined.ld  -T/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../../components/soc/esp32/ld/esp32.peripherals.ld  -Wl,--wrap=k_calloc  -Wl,--wrap=k_malloc  -lnet80211  -lcore  -lpp  -lcoexist  -lphy  -lrtc  -lgcc  -L/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/zephyr/esp32/../blobs/lib/esp32 && cd /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build/zephyr && /nix/store/vh76h64xifdiylk1dfc9s6mgpzazngxs-cmake-3.24.3/bin/cmake -E echo
/nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/12.1.0/../../../../xtensa-espressif_esp32_zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(base64.c.obj): in function `base64_encode':
/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/wpa_supplicant/src/utils/base64.c:166: multiple definition of `base64_encode'; zephyr/libzephyr.a(base64.c.obj):/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/lib/os/base64.c:62: first defined here
/nix/store/zgac4ca0gd1zidn558p6gnbznrrn8bdq-zephyr-sdk-0.15.2/zephyr-sdk-0.15.2/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/12.1.0/../../../../xtensa-espressif_esp32_zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(base64.c.obj): in function `base64_decode':
/home/ricardo/Documents/Programming/zephyr-workspace/modules/hal/espressif/components/wpa_supplicant/src/utils/base64.c:188: multiple definition of `base64_decode'; zephyr/libzephyr.a(base64.c.obj):/home/ricardo/Documents/Programming/zephyr-workspace/zephyr/lib/os/base64.c:126: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /nix/store/vh76h64xifdiylk1dfc9s6mgpzazngxs-cmake-3.24.3/bin/cmake --build /home/ricardo/Documents/Programming/zephyr-workspace/zephyr/build
```

Build of `samles/net/wifi` succeeds without having added `CONFIG_WEBSOCKET_CLIENT=y` to `samples/net/wifi/prj.conf`. 

These are all the steps i took to produce this

--- a/west.yml
+++ b/west.yml
@@ -57,7 +57,7 @@ manifest:
       groups:
         - hal
     - name: hal_espressif
-      revision: a7590d34f28bb1cd3de28847db3f4d31c38ae41d
+      revision: 15c84e32d557545d662392e72ca86e3d88382c03
       path: modules/hal/espressif
       west-commands: west/west-commands.yml
       groups:

